File: 0415.字符串相加/0415-字符串相加.cpp
class Solution {
public:
    string addStrings(string num1, string num2) {
        int m = num1.size();
        int n = num2.size();
        if( m == 0) {
            return num2;
        }
        if( n == 0 ) {
            return num1;
        }

        int i=m-1;
        int j=n-1;
        int k = 0;
        string ans;
        int val = 0;
        int val1 = 0;
        int val2 = 0;

        while( i>=0 || j>=0 || val ) {
            val1 = i>=0 ? num1[i] - '0' : 0 ;
            val2 = j>=0 ? num2[j] - '0' : 0 ;
            val += val1 + val2;
            ans.push_back(val%10 + '0');
            val = val/10;
            i--;
            j--;
        }
        reverse(ans.begin(), ans.end());
        return ans;
    }
};

File: 0415.字符串相加/0415-字符串相加.go
func addStrings(num1 string, num2 string) string {
    ans := ""
    i := len(num1) - 1
    j := len(num2) - 1
    val := 0

    for i>=0 || j>=0 || val >0 {
        val1 := 0
        val2 := 0
        if( i>=0 ) {
            val1 = int(num1[i] - '0')
        }
        if( j>=0 ) {
            val2 = int(num2[j] - '0')
        }
        i--;
        j--;

        val += val1 + val2
        ans = strconv.Itoa(val%10) + ans
        val = val/10
    }

    return ans
}